### PR TITLE
Feat/fetch mp3 url

### PR DESCRIPTION
**Pull request by lionellloh**
1. Given a dev.to URL, I can now parse the html to get the mp3 url 
2. Download the mp3 url. 

String matching logic is not strong or generalisable enough due to my weak sh scripting skills. WIP.

### CODE REVIEW

**Comment by emctackett** [comment]
Very cool, @lionellloh I am able to run this successfully on my machine.

Larson Carter (pod member) is a maintainer for oh-my-zsh and had some concerns/comments about the way we are sourcing the file in our plugin thus far.  This is something you can maybe discuss with him tomorrow but seriously, awesome getting this done today.

**Comment by larson-carter** [comment]
@lionellloh 

Hey, Great work so far on the plugins. I'm excited to see all of the progress that you are going to continue to make on this.

So there are just a few things that I'm going to break down here, however It might be easiest to do over a zoom call.

1. Proper Sourcing - In production and debugging

So [this](https://github.com/ohmyzsh/ohmyzsh/commit/07bcf041c00bb583cddb1b41cdea9f9d221d7d1b) is the proper way on how to source files into omz. However you do not need to modify oh-my-zsh.sh since you will just put your actual plugin itself into a folder that is already being sourced. For this I recommend plugins/ when it is ready to be submitted for the hackathon.

Right now doing "zsh fetch.zsh https://dev.to/iphreaks/ips-292-evolving-apps-and-hacking-around-with-eric-crichlow ./mp3-files2" is fine, however before you submit this for the hackathon it would be best to sneak this into the plugins/ folder, that way it will be sourced for everyone without having to modify some script for the project itself.

Ideally you can run this for testing (if it is not sourced) "zsh fetch.zsh <INSERT URL HERE>"

If you want to shorten the syntax you can simply run these commands:

- source ~/path-to-zsh-file
- fetch <INSERT URL HERE>

2. Directory mixups?

I also am curious as to why you are defining another directory to store the data in your readme you state ./mp3-files2 is where they will be stored. 

But in the actual .zsh file it is defining the folder ./mp3-files? My suggestion for this would to default to ./mp3-files then setup a flag in the script for example -folder /.mp3-files2

If you were to do so then the syntax for testing would be "zsh fetch.zsh <INSERT URL HERE> -folder /.mp3-files"

3. Notes

- Don't bake something together. This will be hard to explain over text so we might need a zoom call so I'm more clear. Basically, don't make a script that depends on other things (I believe that we have been to a hackathon before and seen where people basically polish up an API and make features based on the API - Don't do that) for example don't utilize something that isn't 100% what you need. For example, you don't need to create a DB to track what % of the podcast the user has listened to. That's unnecessary for what you are trying to do. I can go more in detail on this in a zoom call.
- Think logically about people's environments. Not everyone will have all the dependencies that you used installed already.

File: fetch.zsh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+echo "Fetching from $URL ..."
+# MP3_URL=$(curl "$URL" | grep ".mp3" | sed -e 's?https://www\.??' |/#   <source src="/)
+MP3_URL=$(curl "$URL" | grep ".mp3")
+
+# Hackish way for now because I can't do string matching :( 
+MP3_URL=${MP3_URL:17:-20}
+echo "Downloading... $MP3_URL"
+
+# Redirect both std and error output  
+wget $MP3_URL -q --show-progress --progress=bar:force 2>&1 -P $MP3_DIR
```
Comment (larson-carter) [inline fetch.zsh:29]: Just an FYI not every machine has wget installed.

Comment (lionellloh) [reply fetch.zsh:29]: So just to confirm - we will include all the deps and installation instructions in a README.md right? As opposed to some auto-installation mechanism that omz uses.

Comment (larson-carter) [reply fetch.zsh:29]: Yes you will include all deps.

For you guys imagine as if it is going to be in the oh-my-zsh itself. However for judges and me you will need to teach me how to install it.

**Comment by whoiswillma** [comment]
No longer relevant: podcast downloading is handled in python

**Pull request closed without merging**